src/main/java/org/jabref/logic/integrity/BooktitleChecker.java

```diff
@@ -18,6 +30,22 @@ public Optional<String> checkValue(String value) {
             return Optional.of(Localization.lang("booktitle ends with 'conference on'"));
         }
 
+        if (DATE_CHECK.test(value.trim())) {
+            return Optional.of(Localization.lang("Date is contained in the booktitle"));
```

Comment:
When commenting the last time, I totally overlooked that the first match is returned.

Two Options: 1. Change the interace of `checkValue` to return a set of errors. 2. Return a concatenated string. 3. Add new checkers for each apsect of BookTitle.

- Option 1 is bad, because it is only because of this checker
- Option 2 is a hack - easily implemeted by `StringJoiner errors = new StringJoiner(", ");` and then adding to `errors`.
- Option 3 feels best, because it separates the concerns - and the tests are independent from each other (as your regex shows)

Propposal: Split up the checks to different checkers. Call them "PageNumbersContainedChecker", "PersonsContainedChecker", and so on. Enable them on "booktitle" and "journal" fields.

Reply:
Hi @koppor. Thanks for the advice! I've added the checkers as seperate files and written tests for them. I've also adjusted the FieldCheckers file to execute these checkers. Integry checker is now showing mutliple checks which is great! Still working on the location checks but will hopefully be done soon